tests/benchdnn/reorder/reorder.cpp

```diff
@@ -146,6 +146,18 @@ dnnl_status_t init_pd(init_pd_args_t<prb_t> &init_pd_args) {
     for (int d = 0; d < prb->ndims; ++d)
         if (prb->runtime_dim_mask & (1 << d)) dims[d] = DNNL_RUNTIME_DIM_VAL;
 
+    if (is_sycl_engine()) {
+        for (int i = 0; i < prb->ndims; i++) {
+            if (prb->dims[i] > INT_MAX) {
+                BENCHDNN_PRINT(2,
+                        "[CHECK_MEM]: Dimension of size %ld "
+                        "is to big to fit in an int (max %d).\n",
+                        prb->dims[i], INT_MAX);
+                return dnnl_unimplemented;
+            }
+        }
+    }
```

Comment:
This is the library limitation, not benchdnn's. Should be moved from here.

Reply:
Where should it be moved to?

Reply:
Do you mean `init_pd()` in the `reorder.cpp` in benchdnn?

Reply:
I mean `init()` function inside the reorder implementation inside the library. It's a user-visible limitation and the implementation should dispatch into if such dimensions encountered.

Reply:
Done